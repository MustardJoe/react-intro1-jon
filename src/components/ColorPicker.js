import React from 'react';
import styles from './ColorPicker.css';

function ColorPicker() {
  const redHandler = () => {
    /* eslint-disable no-console */
    console.log('red');
  };

  const yellowHandler = () => {
    console.log('yellow');
  };

  const blueHandler = () => {
    console.log('blue');
  };

  return (
    <section className={styles.ColorPicker}>
      <button className={styles.red} onClick={redHandler}>Red</button>
      <button className={styles.yellow} onClick={yellowHandler}>Yellow</button>
      <button className={styles.blue} onClick={blueHandler}>Blue</button>
    </section>
  );
}

export default ColorPicker;

