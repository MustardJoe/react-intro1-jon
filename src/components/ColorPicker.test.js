import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('color picker component my new friend', () => {
  it('renders a ColorPicker', () => {
    const wrapper = shallow(<ColorPicker />);
    expect(wrapper).toMatchSnapshot();
  });

  it('handles the red button press', () => {
    /* eslint-disable no-console */
    console.log = jest.fn();
    const wrapper = shallow(<ColorPicker />);

    wrapper.find('button').at(0).simulate('click');

    /* eslint-disable no-console */
    expect(console.log).toHaveBeenCalledWith('red');
  });

  it('handles the yellow button press', () => {
    /* eslint-disable no-console */
    console.log = jest.fn();
    const wrapper = shallow(<ColorPicker />);

    wrapper.find('button').at(1).simulate('click');

    expect(console.log).toHaveBeenCalledWith('yellow');
  });

  it('handles the blue button press', () => {
    /* eslint-disable no-console */
    console.log = jest.fn();
    const wrapper = shallow(<ColorPicker />);

    wrapper.find('button').at(2).simulate('click');

    expect(console.log).toHaveBeenCalledWith('blue');
  });
});
