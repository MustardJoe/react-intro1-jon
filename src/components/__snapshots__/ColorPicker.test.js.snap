// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color picker component my new friend renders a ColorPicker 1`] = `
<section
  className="ColorPicker"
>
  <button
    className="red"
    onClick={[Function]}
  >
    Red
  </button>
  <button
    className="yellow"
    onClick={[Function]}
  >
    Yellow
  </button>
  <button
    className="blue"
    onClick={[Function]}
  >
    Blue
  </button>
</section>
`;
